### DISCUSSION TITLE
LED not working as expected

### DISCUSSION THREAD

**Discussion by Belcraig** [klaasnicolaas/home-assistant-glow#163] [Q&A]
Hi all,
I've very little experience but think I have it all set up correctly but there are a few things that are not working as i'm expecting them to.
Firstly the LED, 
I have the yaml below and reading the logs, i can see that it is registering the output and i can see in HA that it is getting readings, (abeit high ones - I seen a different topic for this i'll look at), however the LED does not flash (or actually respond).
I've got it connect up in line with the table, but i'm not sure what i'm missing or how why its not working.

https://imgur.com/huw6jZH
on my board i got (ESP32)
 
R - GPIO2
G - GPIO4
B - GPIO5
and the - going to a GND

https://imgur.com/4t1e8IF

shows orientation on LED. 

---
    # Home Assistant Glow
    #
    # Read your electricity meter by means of the pulse LED on your 
    # meter, useful if you do not have a serial port (P1).
    #
    substitutions:
      device_name: home-assistant-glow
      friendly_name: House
      project_version: "3.0.0"
      device_description: "Measure your energy consumption with the pulse LED on your smart meter"
      pulse_pin: GPIO12
      status_led: GPIO5
    
      # imp/kWh rate ⬇ #
      pulse_rate: '1000'
    
      # Webserver credentials ⬇ #
      web_username: 'glow'
      web_password: 'esphome'
    
    dashboard_import:
      package_import_url: github://klaasnicolaas/home-assistant-glow/home_assistant_glow.yaml
    
    esphome:
      name: '${device_name}'
      comment: '${device_description}'
      project:
        name: "klaasnicolaas.home-assistant-glow"
        version: "${project_version}"
    
    # Choose the right Platform
    # esp32: https://esphome.io/components/esp32.html
    # esp8266: https://esphome.io/components/esp8266.html
    esp32:
      board: esp32dev
    
    # WiFi credentials #
    wifi:
      ssid: !secret wifi_ssid
      password: !secret wifi_password
    
      # Enable fallback hotspot (captive portal) in case wifi connection fails
      ap:
        ssid: '${device_name}'
    
    captive_portal:
    
    # Enable logging
    logger:
    
    # Enable improv over serial
    improv_serial:
    
    # Enable Home Assistant API
    api:
      services:
        - service: reset_total_energy
          then:
            - button.press:
                id: button_reset_total
    
    ota:
      safe_mode: true
      reboot_timeout: 10min
      num_attempts: 5
    
    web_server:
      port: 80
      auth:
        username: '${web_username}'
        password: '${web_password}'
    
    output:
      # - platform: gpio
      #   pin: GPIO5
      #   id: output_blue
      - platform: gpio
        pin: GPIO2
        id: output_red
      - platform: gpio
        pin: GPIO4
        id: output_green
    
    light:
      - platform: binary
        internal: true
        id: led_red
        name: Red
        output: output_red
    
    # Status LED for connection
    status_led:
      pin:
        # Blue LED
        number: ${status_led}
    
    button:
      - platform: restart
        name: "Restart - Glow"
      - platform: template
        name: "Reset - Total Energy"
        id: button_reset_total
        on_press:
          - pulse_meter.set_total_pulses:
              id: sensor_energy_pulse_meter
              value: 0
    
    # Sensors for ESP version and WIFI information
    text_sensor:
      # Installed version
      - platform: template
        name: "Glow - Installed version"
        id: glow_version
        icon: "mdi:label-outline"
        entity_category: diagnostic
        lambda: |-
          return {"${project_version}"};
        update_interval: 6h
      - platform: version
        hide_timestamp: true
        name: '${friendly_name} - ESPHome Version'
      - platform: wifi_info
        ip_address:
          name: '${friendly_name} - IP Address'
          icon: mdi:wifi
        ssid:
          name: '${friendly_name} - Connected SSID'
          icon: mdi:wifi-strength-2
    
    sensor:
      # WiFi signal
      - platform: wifi_signal
        name: "${friendly_name} - WiFi Signal"
        update_interval: 120s
    
      # Pulse meter
      - platform: pulse_meter
        name: '${friendly_name} - Power Consumption'
        id: sensor_energy_pulse_meter
        unit_of_measurement: 'W'
        state_class: measurement
        device_class: power
        icon: mdi:flash-outline
        accuracy_decimals: 0
        pin: ${pulse_pin}
        # internal_filter: 100ms
        on_value:
          then:
            - light.turn_on:
                id: led_red
            - delay: 0.5s
            - light.turn_off:
                id: led_red
        filters:
          # multiply value = (60 / imp value) * 1000
          # - multiply: 60
          - lambda: return x * ((60.0 / ${pulse_rate}) * 1000.0);
        total:
          name: '${friendly_name} - Total Energy'
          id: sensor_total_energy
          unit_of_measurement: 'kWh'
          icon: mdi:circle-slice-3
          state_class: total_increasing
          device_class: energy
          accuracy_decimals: 3
          filters:
            # multiply value = 1 / imp value
            # - multiply: 0.001
            - lambda: return x * (1.0 / ${pulse_rate});
      # Total day useage
      - platform: total_daily_energy
        name: '${friendly_name} - Daily Energy'
        id: sensor_total_daily_energy
        power_id: sensor_energy_pulse_meter
        unit_of_measurement: 'kWh'
        icon: mdi:circle-slice-3
        state_class: total_increasing
        device_class: energy
        accuracy_decimals: 3
        filters:
          # Multiplication factor from W to kW is 0.001
          - multiply: 0.001
    
    # Enable time component to reset energy at midnight
    # https://esphome.io/components/time.html#home-assistant-time-source
    time:
      - platform: homeassistant
        id: homeassistant_time

Testing the LED with a coin battery or connecting the - to the 3v3 on the other side lights the LED so i know its not faulty.

any steps i can try?

thanks in advance for your help.

**Comment by klaasnicolaas** [comment]
Good to know that the LED does work, doesn't it happen to have a common `+` instead of `-`? See also: https://github.com/klaasnicolaas/home-assistant-glow/discussions/112

**Reply by Belcraig** [reply]
saw that and wasn't sure of the difference in the wiring set up. but the longest leg is attached to the biggest section of the LED making this a (-)?
if it is a (+) where would the GND wire go, or is it a case of needing a different LED needed?

**Reply by Belcraig** [reply]
just incase i'm getting it wrong here the diode is working fine, the corresponding LED flash that isn't working...

**Reply by klaasnicolaas** [reply]
The difference is that you connect the common pin to a `+` or `-` depending on the type of LED you have. In the documentation I assume that you order a common cathode, but the anode version is also fine as long as you adjust the wiring to it.

And as suggested in the other discussion, you can use a coin battery to find out what the correct polarity is and whether you are dealing with cathode or anode.
